packages/material-ui/src/Slider/Slider.js

```diff
@@ -566,6 +570,10 @@ const Slider = React.forwardRef(function Slider(props, ref) {
   };
 
   const handleTouchMove = useEventCallback((nativeEvent) => {
+    if (disabled) {
+      return;
+    }
```

Comment:
I don't think that this is a proper solution, the touch move will leak until the component gets unmounted. What about we add disabled in the list of dependencies for the event cleanup logic?

Reply:
Still pretty new to React, but I think I'm doing this now for the mouse. Not sure if you wanted it done for the keyboard as well/how to do it for the keyboard as well, but I can look into it if that's the direction you think it should go. Looking at the sandbox you posted above, it does look like firefox prevents the keyboard from making changes to the ranged input even without the blur.